include/task_manager.h

```diff
@@ -74,6 +75,7 @@ namespace detail {
 				prepass_handler cgh(tid, std::make_unique<command_group_storage<CGF>>(cgf), num_collective_nodes);
 				cgf(cgh);
 				task& task_ref = register_task_internal(std::move(reservation), std::move(cgh).into_task());
+				tsk_ptr = &task_ref;
 
 				compute_dependencies(tid);
```

Comment:
Maybe this should also directly take tasks? The other call-site could do this as well (if `reduce_execution_front` return a task rather than an id, which it probably should).

Reply:
Right, I've changed the task manager to also pass around references instead of ids in most places.